server/common/server.js

```diff
@@ -34,25 +34,23 @@ export default class ExpressServer {
     return this;
   }
 
-  listen(port = process.env.PORT) {
+  async listen(port = process.env.PORT) {
     const welcome = (p) => () =>
       l.info(
         `up and running in ${process.env.NODE_ENV ||
           "development"} @: ${os.hostname()} on port: ${p}}`
       );
 
-    oas(app, this.routes)
-      .then(() => {
-        mongo().then(() => {
-          l.info(`Database loaded!`);
-          http.createServer(app).listen(port, welcome(port));
-        });
-      })
-      .catch((e) => {
-        l.error(e);
-        exit(1);
-      });
-
+    try {
```

Comment:
Hey @manishprivet!

There is a small issue here. If you check server/index.js, we're exporting the express app from that file. But since you made the listen function to be async, it will export a promise instead of the app variable which is causing tests to fail.

Here's what you can do:-
1. Make a separate function called load() which will await for oas and mongo and call the welcome function.
2. Make the listen function sync again.

Sample:-

async load(port) {
    const welcome = (p) => () =>
      l.info(
        `up and running in ${process.env.NODE_ENV ||
          "development"} @: ${os.hostname()} on port: ${p}}`
      );

    await oas(app, this.routes);
    await mongo();

    l.info(`Database loaded!`);
    http.createServer(app).listen(port, welcome(port));
  }

  listen(port = process.env.PORT) {
    try {
      this.load(port);
    } catch (err) {
      l.error(err);
      exit(1);
    }

    return app;
  }